block/produce.go

```diff
@@ -128,6 +133,38 @@ func (m *Manager) produceApplyGossip(ctx context.Context, allowEmpty bool, nextP
 	return block, commit, nil
 }
 
+// SnapshotSequencerSet loads two versions of the sequencer set:
+//   - the one that was used for the last block (from the store)
+//   - and the most recent one (from the manager memory)
+//
+// It then calculates the diff between the two and creates consensus messages for the new sequencers,
+// i.e., only for the diff between two sets.  The new sequencer set will be used for next block and
+// will be stored in the state after the block production instead of the old sequencer set.
+func (m *Manager) SnapshotSequencerSet() (allSequencers types.Sequencers, err error) {
+	// the most recent sequencer set
+	allSequencers = m.Sequencers.GetAll()
+
+	// the sequencer set that was used for the last block
+	lastSequencers, err := m.Store.LoadLastBlockSequencerSet()
```

Comment:
@keruch what the expected behavior after rotation? the new sequencer won't have `LastBlockSequencerSet` in store

Reply:
added a comment
```
// The set from the state is dumped to memory on reboots. It helps to avoid sending unnecessary
// UspertSequencer consensus messages on reboots. This is not a 100% solution, because the sequencer set
// is not persisted in the store in full node mode. It's only used in the proposer mode. Therefore,
// on rotation from the full node to the proposer, the sequencer set is duplicated as consensus msgs.
// Though single-time duplication it's not a big deal.
```